packages/system/src/Stock/ListView/api.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ListApi,
+  SortBy,
+  ItemSortFieldInput,
+  OmSupplyApi,
+  FilterBy,
+} from '@openmsupply-client/common';
+import { availableBatchesGuard, itemsGuard } from '../../Item/utils';
+import { StockRow } from '../types';
+const onRead =
+  (api: OmSupplyApi) =>
+  async ({
+    sortBy,
+    filterBy,
+  }: {
+    first: number;
+    offset: number;
+    sortBy: SortBy<StockRow>;
+    filterBy: FilterBy<StockRow> | null;
+  }): Promise<{
+    nodes: StockRow[];
+    totalCount: number;
+  }> => {
+    const result = await api.itemsWithStockLines({
+      first: 1000,
+      offset: 0,
+      key: ItemSortFieldInput.Name,
+      desc: sortBy.isDesc,
+      filter: filterBy,
+    });
+
+    const items = itemsGuard(result);
+    const nodes: StockRow[] = [];
+    items.nodes.forEach(item => {
+      const availableBatches = availableBatchesGuard(item.availableBatches);
+      availableBatches
+        .filter(batch => batch.totalNumberOfPacks > 0)
+        .forEach(batch =>
```

Comment:
just for the future.. I think showing items with no stock is *probably* helpful?

Reply:
probably is - we don't currently, you have to go through items and check the stock on hand there. really, this whole list could be replaced by a toggle or filter on the item list 'only show items with stock on hand' and maybe a view option which shows a row per batch. but hey.. this is useful for now